src/services/AnalysisService.ts

```diff
@@ -219,19 +390,34 @@ class AnalysisService {
 
     if (contributingDeveloperAudit?.length === 0) {
       soosLogger.info(`Integration Name: ${integrationName}`);
-      const envVariableName = integrationNameToEnvVariable[integrationName];
-      if (envVariableName) {
-        const contributingDeveloper = process.env[envVariableName];
-        if (contributingDeveloper) {
+
+      // loop through all possible contributing developer environment variables and add the values for any that are found
+      for (const ev in contributingDeveloperEnvironmentVariables) {
+        const environmentVariableValue = process.env[ev];
+
+        if (environmentVariableValue) {
           contributingDeveloperAudit.push({
             source: ContributingDeveloperSource.EnvironmentVariable,
-            sourceName: envVariableName,
-            contributingDeveloperId: contributingDeveloper,
+            sourceName: ev,
+            contributingDeveloperId: environmentVariableValue,
           });
         }
       }
     }
 
+    commitHash = commitHash
+      ? commitHash
+      : this.findFirstEnvironmentVariableValue(commitHashEnvironmentVariables);
+    branchName = branchName
+      ? branchName
+      : this.findFirstEnvironmentVariableValue(branchNameEnvironmentVariables);
+    buildUri = branchUri
+      ? branchUri
+      : this.findFirstEnvironmentVariableValue(buildUriEnvironmentVariables);
+    buildVersion = buildVersion
+      ? buildVersion
+      : this.findFirstEnvironmentVariableValue(buildNumberEnvironmentVariables);
```

Comment:
are these null or undefined? then you can do
```suggestion
    commitHash = commitHash ?? this.findFirstEnvironmentVariableValue(commitHashEnvironmentVariables);
    branchName = branchName ?? this.findFirstEnvironmentVariableValue(branchNameEnvironmentVariables);
    buildUri = branchUri ?? this.findFirstEnvironmentVariableValue(buildUriEnvironmentVariables);
    buildVersion = buildVersion ?? this.findFirstEnvironmentVariableValue(buildNumberEnvironmentVariables);
```

Reply:
null, undefined or maybe ''